Assets/Scripts/Models/Need.cs

```diff
@@ -83,25 +89,57 @@ public string DisplayAmount
         }
     }
 
+    /// <summary>
+    /// Gets the EventAction for the current furniture.
+    /// These actions are called when an event is called. They get passed the furniture
+    /// they belong to, plus a deltaTime (which defaults to 0).
+    /// </summary>
+    /// <value>The event actions that is called on update.</value>
+    public EventActions EventActions { get; private set; }
+
     // Update is called once per frame
     public void Update(float deltaTime)
     {
-        FunctionsManager.Need.CallWithInstance(luaUpdate, this, deltaTime);
-        if (luaOnly)
+        if (EventActions != null && EventActions.HasEvent("OnUpdate"))
         {
-            return;
+            EventActions.Trigger("OnUpdate", this, deltaTime);
         }
-
-        Amount += growthRate * deltaTime;
-
-        if (character != null && character.CurrTile.GetGasPressure("O2") < 0.15)
+        else 
         {
-            Amount += (addedInVacuum - (addedInVacuum * (character.CurrTile.GetGasPressure("O2") * 5))) * deltaTime;
+            DefaultNeedDecay(deltaTime);
         }
 
         if (Amount.AreEqual(100))
         {
-            // FIXME: Insert need fail damage code here.
+            if (EventActions != null && EventActions.HasEvent("OnEmptyNeed"))
+            {
+                EventActions.Trigger("OnEmptyNeed", this, deltaTime);
+            }
+            else
+            {
+                DefaultEmptyNeed();
+            }
+        } 
+        else if (Amount > 90f)
+        {
+            if (EventActions != null && EventActions.HasEvent("OnSevereNeed"))
```

Comment:
Trigger does nothing if it doesn't have that action. Maybe we don't even need to check if it exists. And if we assume that EventActions is always present, we could remove the extra ifs.